File: batch11amsm/FirstApplication/app/src/main/java/com/example/firstapplication/fragments/FirstFragment.java
package com.example.firstapplication.fragments;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;

import com.example.firstapplication.R;


public class FirstFragment extends Fragment {

Button movebtn;
EditText emailedit;



    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_first, container, false);
    }

    @Override
    public void onViewCreated( View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        movebtn = view.findViewById(R.id.movetosecond);
     emailedit = view.findViewById(R.id.editmail);
        movebtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
             String emailtext = emailedit.getText().toString();
             Fragment fragment = new SecondFragment();
             Bundle bundle = new Bundle();
             bundle.putString("data",emailtext);
             fragment.setArguments(bundle);


                FragmentManager fragmentManager = getParentFragmentManager();
                FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
                fragmentTransaction.replace(R.id.fragmentcontainer,fragment);
                fragmentTransaction.commit();
            }
        });
    }
}

File: kushalapp/BasickComponent/app/src/main/java/com/cbitss/basickcomponent/fragments/FirstFragment.kt
package com.cbitss.basickcomponent.fragments

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import com.cbitss.basickcomponent.R


class FirstFragment : Fragment() {

lateinit var movebtn:Button


    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_first, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        movebtn =view.findViewById(R.id.movetosecond)

        movebtn.setOnClickListener {

            val email = "abc@gmail.com"
             val fragment = SecondFragment()
             val bundle = Bundle()
            bundle.putString("myemail",email)
            fragment.arguments = bundle

            val fragmentmanager = parentFragmentManager
            val fragmenttransction = fragmentmanager.beginTransaction()
            fragmenttransction.replace(R.id.fragmentcontainer,fragment)
            fragmenttransction.commit()
        }
    }
}